runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SerializablePipelineOptions.java

```diff
@@ -17,35 +17,68 @@
  */
 package org.apache.beam.runners.core.construction;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import java.io.IOException;
 import java.io.ObjectInputStream;
 import java.io.Serializable;
 import org.apache.beam.sdk.io.FileSystems;
 import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.util.common.ReflectHelpers;
 
 /**
  * Holds a {@link PipelineOptions} in JSON serialized form and calls {@link
  * FileSystems#setDefaultPipelineOptions(PipelineOptions)} on construction or on deserialization.
  */
 public class SerializablePipelineOptions implements Serializable {
+  private static final ObjectMapper MAPPER =
+      new ObjectMapper()
+          .registerModules(ObjectMapper.findModules(ReflectHelpers.findClassLoader()));
 
   private final String serializedPipelineOptions;
   private transient PipelineOptions options;
 
   public SerializablePipelineOptions(PipelineOptions options) {
-    this.serializedPipelineOptions = PipelineOptionsSerializationUtils.serializeToJson(options);
+    this.serializedPipelineOptions = serializeToJson(options);
     this.options = options;
     FileSystems.setDefaultPipelineOptions(options);
   }
 
+  public SerializablePipelineOptions(String json) {
```

Comment:
Please, add a unit test for this constructor. 
Also, we need to make sure that two objects created by different constructors (from `PipelineOptions` and `String`) are equal.

Reply:
Thanks for the suggestion ! In case serialize from/to json is not bijective, we might end up having different PipelineOptions, you're right

Reply:
```
    PipelineOptions options = PipelineOptionsFactory.create();
    PipelineOptions options2 = PipelineOptionsFactory.create();
    assertTrue(options.equals(options2));
    assertTrue(Objects.equals(options, options2));
```
both assert fail ! That is because pipelineOptions is a proxy, its equal impl compares the references. So in my SerializablePipelineOptions.equals() impl (usefull to test objects created with the 2 constructors), I compare only the json strings.